packages/@aws-cdk/aws-appmesh/lib/virtual-node.ts

```diff
@@ -191,6 +191,10 @@ export class VirtualNode extends VirtualNodeBase {
     const accessLogging = props.accessLog?.bind(this);
     const serviceDiscovery = props.serviceDiscovery?.bind(this);
 
+    if ((props.listeners && props.listeners.length > 0) && !serviceDiscovery) {
+      throw new Error('If you specify a listener, then you must specify service discovery information.');
+    };
```

Comment:
1. Hmm, can you explain why is this validation added? This is a breaking change, no?
2. This is actually wrong - you can also specify a listener using the `addListener()` method, and you don't check for that. I believe you want to add this validation to the `validate()` method.

Reply:
> Hmm, can you explain why is this validation added? This is a breaking change, no?

This is the restriction on `serviceName` that is mentioned in [this doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-virtualnodespec.html#cfn-appmesh-virtualnode-virtualnodespec-servicediscovery). I don't think it is a breaking change.

> This is actually wrong - you can also specify a listener using the addListener() method,

Great point. I totally overlooked this.

> I believe you want to add this validation to the validate() method.

Well noted. Will add `validate()` function.

Reply:
> > Hmm, can you explain why is this validation added? This is a breaking change, no?
> 
> This is the restriction on `serviceName` that is mentioned in [this doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-virtualnodespec.html#cfn-appmesh-virtualnode-virtualnodespec-servicediscovery). I don't think it is a breaking change.

What you're saying is that this would have failed at deploy time before your change if a `listener` was specified without the `serviceDiscovery` property - am I understanding this correctly?

Reply:
> Hmm, can you explain why is this validation added? This is a breaking change, no?
>>This is the restriction on serviceName that is mentioned in this doc. I don't think it is a breaking change.
>>>What you're saying is that this would have failed at deploy time before your change if a listener was specified without the serviceDiscovery property - am I understanding this correctly?

Yes, you are correct!

Reply:
> > > Hmm, can you explain why is this validation added? This is a breaking change, no?
> > 
> > 
> > This is the restriction on `serviceName` that is mentioned in [this doc](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-appmesh-virtualnode-virtualnodespec.html#cfn-appmesh-virtualnode-virtualnodespec-servicediscovery). I don't think it is a breaking change.
> 
> What you're saying is that this would have failed at deploy time before your change if a `listener` was specified without the `serviceDiscovery` property - am I understanding this correctly?

Yes, you are correct!

Reply:
Ok, thanks for the explanation. Ideally, we would have added this in a separate PR, because it seems unrelated to the other changes in this PR. But I'm fine leaving it here in this case.

Reply:
Thanks and that makes sense.
Moving on, will do my best to stick with specific feature to create PR.